debezium-api/src/main/java/io/debezium/annotation/Incubating.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+/**
+ * Marks the annotated element as incubating. The contract of incubating elements (e.g. packages, types, methods,
+ * constants etc.) is under active development and may be incompatibly altered - or removed - in subsequent releases.
+ * <p>
+ * Usage of incubating API/SPI members is encouraged (so the development team can get feedback on these new features)
+ * but you should be prepared for updating code which is using them as needed.
+ *
+ */
+@Documented
+@Retention(RetentionPolicy.CLASS)
+public @interface Incubating {
```

Comment:
@gunnarmorling Thanks that you've noticed :-). This was intended to lead to another question, what other annotations should move from core to API? Also should annotation like `@ThreadSafe` be used on the `DebeziumEnginre` as it is more implementation detail?

Reply:
Good question, from a quick glance I'd say `@Incubating` is the only one that's really user-facing. What I'm realizing now though is the fact that there's a split package between core and api now. That's something we should avoid, so to avoid issues with the JPMS in some future.

Reply:
@gunnarmorling The only shared package is the `annotation` so we can use just a different name. How about using `documenting` as it does not have any real function and is just for documentation?

Reply:
@gunnarmorling Package renamed